packages/flutter_tools/lib/src/run_cold.dart

```diff
@@ -125,22 +125,29 @@ class ColdRunner extends ResidentRunner {
   @override
   void printHelp({ @required bool details }) {
     bool haveDetails = false;
+    bool haveAnything = false;
     for (FlutterDevice device in flutterDevices) {
       final String dname = device.device.name;
       if (device.observatoryUris != null) {
-        for (Uri uri in device.observatoryUris)
+        for (Uri uri in device.observatoryUris) {
           printStatus('An Observatory debugger and profiler on $dname is available at $uri');
+          haveAnything = true;
+        }
       }
     }
     if (supportsServiceProtocol) {
       haveDetails = true;
-      if (details)
+      if (details) {
         printHelpDetails();
+        haveAnything = true;
```

Comment:
Should this not consult the return value of `printHelpDetails()`?

Reply:
oh I meant to remove the return value. Turns out to do it strictly correct you'd need to be able to get the return value without calling it (or we could change it to have it return a string or list of strings instead of printing), and in practice it doesn't really matter that much since we can predict how it will react in the cases that matter.